### ISSUE TITLE
VSCode + Hosted WASM: Can we debug WASM while also rebuilding server project?

### ISSUE THREAD

**Issue by matthewcorven** [dotnet/AspNetCore.Docs#20934]
Lacks launch profile example allowing IDE-side debugging of WASM while also rebuilding the server project.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ec2f61c7-9f51-6735-e328-3c7b495d70c3
* Version Independent ID: 4d06e565-282d-6387-aa90-cb2c2986d35a
* Content: [Debug ASP.NET Core Blazor WebAssembly](https://docs.microsoft.com/en-us/aspnet/core/blazor/debug?view=aspnetcore-3.1&tabs=visual-studio#launch-configuration-options)
* Content Source: [aspnetcore/blazor/debug.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/blazor/debug.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
Thanks for the feedback! I'm buried in work ⛏️⛰️ for the next couple of days then out for an extended ⛄ *holiday vacation* 🎁. I've added this to my **Post-holiday Triage** issue list and will take a look when I return. Sorry for the delay, but I must get some long-standing priorities resolved before leaving, so only bug fixes will usually be addressed before I get some *much needed* rest 😅. I assure you that this issue won't get lost.

**Comment by guardrex** [comment]
The product unit indicated that the default `launch.json` file is all that's needed to make the scenario work.

https://docs.microsoft.com/en-us/aspnet/core/blazor/debug?view=aspnetcore-3.1&tabs=visual-studio-code#debug-hosted-blazor-webassembly

If it doesn't work for you, you might want to close here and open an issue for the product unit at ...

https://github.com/dotnet/aspnetcore/issues

... since we can't troubleshoot a problem here on the docs repo.

**Comment by guardrex** [comment]
@captainsafia ... I think this requires you to comment on "while also rebuilding the server project." I'm not sure what it means, and the coverage seems complete to me for hosted Blazor WASM.

**Comment by matthewcorven** [comment]
Perhaps some context would be helpful.  

While focused primarily on the frontend and debugging UI-specific code, it's still necessary to tweak the backend from time to time and I'd prefer it to rebuild on its own (think `dotnet watch`) so that I don't need to stop my train of thought.

**Comment by guardrex** [comment]
I'm back from vacation ...... a *stay-cation* :cry:.

@captainsafia ... AFAICT, this is more of a product unit issue than a docs issue. Do you want to move it over or address/close it here?

**Comment by captainsafia** [comment]
@guardrex @matthewcorven Thanks for your patience.

So...unfortunately...these two things don't work readily with the current experience. The WebSocket connection between the Chrome browser and the debugging proxy on the application service behaves unreliably in this scenario.

I tried tinkering to see if there was a way to use a standard JS debugger config with a `dotnet watch run` process spawn on the CLI. Something like:

```
{
            "name": ".NET Core Debug Blazor Web Assembly in Chrome",
            "type": "pwa-chrome",
            "request": "launch",
            "timeout": 30000,
            "url": "https://localhost:5001",
            "inspectUri": "{wsProtocol}://{url.hostname}:{url.port}/_framework/debug/ws-proxy?browser={browserInspectUri}"
 }
```

...but this didn't prove very effective on my setup.

We're hoping to improve the overall inner loop for .NET 6 in the future. We've got several issues tracking this over on the aspnetcore repo so we shouldn't need to create a new bug there.

Hopefully, we'll have a more helpful solution for this in the future!
